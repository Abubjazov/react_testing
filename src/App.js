import { useEffect, useState } from 'react'

import './App.css'

export const App = () => {
  const [data ,setData] = useState(null)

  useEffect(() => {
    const stOne = setTimeout(() => {
      setData({})
    }, 1000)

    return () => {
      clearTimeout(stOne)
    }
  }, [])

  return (
    <div className='App'>
      {data && <div>DATA</div>}
      <h1>Goodby world</h1>
      <button>Kick me!!!</button>
      <input type={'text'} placeholder={'input value...'}></input>
    </div>
  )
}
