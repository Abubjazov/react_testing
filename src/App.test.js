import { render, screen } from '@testing-library/react'
import { App } from './App'

test('render App - root component', () => {
  render(<App />)

  const gwElem = screen.getByText(/goodby world/i)
  const btn = screen.getByRole('button')
  const input = screen.getByPlaceholderText(/input value/i)

  expect(gwElem).toBeInTheDocument()
  expect(btn).toBeInTheDocument()
  expect(input).toBeInTheDocument()

  expect(input).toMatchSnapshot()

  screen.debug()
})
