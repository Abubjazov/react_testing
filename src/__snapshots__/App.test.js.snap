// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render App - root component 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;
